pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -330,31 +325,105 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 	return nil
 }
 
-func updateStatusFromBuildStatus(taskRun *v1alpha1.TaskRun, buildStatus buildv1alpha1.BuildStatus) {
-	if buildStatus.GetCondition(duckv1alpha1.ConditionSucceeded) != nil {
-		taskRun.Status.SetCondition(buildStatus.GetCondition(duckv1alpha1.ConditionSucceeded))
-	} else {
-		// If the buildStatus doesn't exist yet, it's because we just started running
+func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod) {
+	fmt.Println("taskrun.Status", taskRun.Status)
+	fmt.Println("taskrun.Status.ConditionSucceed", taskRun.Status.GetCondition(duckv1alpha1.ConditionSucceeded))
+
+	if taskRun.Status.GetCondition(duckv1alpha1.ConditionSucceeded) == nil || taskRun.Status.GetCondition(duckv1alpha1.ConditionSucceeded).Status == corev1.ConditionUnknown {
+		// If the taskRunStatus doesn't exist yet, it's because we just started running
 		taskRun.Status.SetCondition(&duckv1alpha1.Condition{
 			Type:    duckv1alpha1.ConditionSucceeded,
 			Status:  corev1.ConditionUnknown,
 			Reason:  reasonRunning,
 			Message: reasonRunning,
 		})
 	}
-	taskRun.Status.StartTime = buildStatus.StartTime
-	if buildStatus.Cluster != nil {
-		taskRun.Status.PodName = buildStatus.Cluster.PodName
-	}
-	taskRun.Status.CompletionTime = buildStatus.CompletionTime
+
+	taskRun.Status.StartTime = &pod.CreationTimestamp
+	taskRun.Status.PodName = pod.Name
 
 	taskRun.Status.Steps = []v1alpha1.StepState{}
-	for i := 0; i < len(buildStatus.StepStates); i++ {
-		state := buildStatus.StepStates[i]
-		taskRun.Status.Steps = append(taskRun.Status.Steps, v1alpha1.StepState{
-			ContainerState: *state.DeepCopy(),
+	for _, s := range pod.Status.ContainerStatuses {
+		if s.State.Terminated != nil {
+			taskRun.Status.Steps = append(taskRun.Status.Steps, v1alpha1.StepState{
+				ContainerState: *s.State.DeepCopy(),
+			})
+		}
+	}
+
+	switch pod.Status.Phase {
+	case corev1.PodRunning:
+		taskRun.Status.SetCondition(&duckv1alpha1.Condition{
+			Type:   duckv1alpha1.ConditionSucceeded,
+			Status: corev1.ConditionUnknown,
+			Reason: "Building",
```

Comment:
maybe `Running` instead of `Building`?

Reply:
I didn't change what we had before when using `BuildStatus` but yeah I agree.